linux/keyman-config/tests/test_handle_install.py

```diff
@@ -0,0 +1,106 @@
+
+    @patch('keyman_config.handle_install._install_package')
+    def test_downloadAndInstallPackage_fileUrl(self, installPackageMethod):
+        # Execute
+        download_and_install_package('file:///tmp/keyboard/sil_euro_latin.kmp')
+
+        # Verify
+        installPackageMethod.assert_called_with('/tmp/keyboard/sil_euro_latin.kmp', '')
+
+    @patch('keyman_config.handle_install._install_package')
+    def test_downloadAndInstallPackage_fileUrlWithBcp47(self, installPackageMethod):
+        # Execute
+        download_and_install_package('file:///tmp/keyboard/sil_euro_latin.kmp?bcp47=dyo')
+
+        # Verify
+        installPackageMethod.assert_called_with('/tmp/keyboard/sil_euro_latin.kmp', 'dyo')
+
+    @patch('keyman_config.handle_install._install_package')
+    def test_downloadAndInstallPackage_fileUrlWithBcp47AndVersion(self, installPackageMethod):
+        # Execute
+        download_and_install_package('file:///tmp/keyboard/sil_euro_latin.kmp?bcp47=dyo&version=1')
+
+        # Verify
+        installPackageMethod.assert_called_with('/tmp/keyboard/sil_euro_latin.kmp', 'dyo')
+
+    @patch('keyman_config.handle_install._install_package')
+    def test_downloadAndInstallPackage_InvalidUrl(self, installPackageMethod):
+        # Execute
+        download_and_install_package('http://localhost/keyboard/sil_euro_latin.kmp')
+
+        # Verify
+        installPackageMethod.assert_not_called()
+
+    @patch('keyman_config.handle_install._install_package')
+    @patch('keyman_config.get_kmp.download_kmp_file')
+    def test_downloadAndInstallPackage_invalidUrl(self, downloadKmpFileMethod,
+                                                  installPackageMethod):
+        for url in ['foo://download/keyboard/sil_euro_latin', 'keyman://keyboard/sil_euro_latin',
+                    'keyman://download/sil_euro_latin', 'keyman://download/keyboard/']:
+            with self.subTest(url=url):
+                # Execute
+                download_and_install_package(url)
+
+                # Verify
+                downloadKmpFileMethod.assert_not_called()
+                installPackageMethod.assert_not_called()
+
+    @patch('keyman_config.handle_install._install_package')
+    @patch('keyman_config.get_kmp.keyman_cache_dir')
+    @patch('keyman_config.handle_install.download_kmp_file')
+    def test_downloadAndInstallPackage_keymanUrl(self, downloadKmpFileMethod,
+                                                 keymanCacheDirMethod, installPackageMethod):
+        # Setup
+        mockPackagePath = '/tmp/sil_euro_latin'
+        keymanCacheDirMethod.return_value = '/tmp'
+        downloadKmpFileMethod.return_value = mockPackagePath
+
+        # Execute
+        download_and_install_package('keyman://download/keyboard/sil_euro_latin')
+
+        # Verify
+        downloadKmpFileMethod.assert_called_with(
+            KeymanComUrl + '/go/package/download/sil_euro_latin?platform=linux&tier=alpha',
```

Comment:
This assertion will fail for beta and stable builds?

Reply:
done